synapse/api/auth.py

```diff
@@ -620,3 +621,115 @@ async def check_user_in_room_or_world_readable(
 
     async def check_auth_blocking(self, *args, **kwargs) -> None:
         await self._auth_blocking.check_auth_blocking(*args, **kwargs)
+
+    async def is_room_visible(
```

Comment:
> Curious why this was changed from accessible to visible?

I felt like accessible suggested joinable, which may not be the case if it is only history_visisbility allowing you to see it but it is otherwise invite-only.

> Is there any changes from SpaceSummary._is_local_room_accessible or was it copied verbatim?

Verbatim other than changing the docstring/comment to not talk about summaries

Reply:
It feels like a fundamental & reusable implementation of MSC3173 which can be reused anywhere, so will keep it extracted to api.auth

Reply:
I think I'd be keen to keep this where it was to simplify this PR.